crates/hulk_replayer/src/window.rs

```diff
@@ -12,56 +12,83 @@ use crate::{
     coordinate_systems::{AbsoluteTime, FrameRange, RelativeTime, ViewportRange},
     labels::Labels,
     timeline::Timeline,
+    worker_thread::PlayerState,
 };
 
 pub struct Window {
-    time_sender: watch::Sender<SystemTime>,
+    time_sender: watch::Sender<PlayerState>,
     frame_range: FrameRange,
     viewport_range: ViewportRange,
-    position: RelativeTime,
     indices: BTreeMap<String, Vec<Timing>>,
 }
 
 impl Window {
     pub fn new(
         indices: BTreeMap<String, Vec<Timing>>,
-        time_sender: watch::Sender<SystemTime>,
+        time_sender: watch::Sender<PlayerState>,
     ) -> Self {
         let frame_range = join_timing(&indices);
         let viewport_range = ViewportRange::from_frame_range(&frame_range);
+        time_sender.send_modify(|state| state.time = frame_range.start().inner());
 
         Self {
             time_sender,
             frame_range,
             viewport_range,
-            position: RelativeTime::new(0.0),
             indices,
         }
     }
 
-    fn replay_at_position(&mut self) {
-        let timestamp = self.position.map_to_absolute_time(&self.frame_range);
+    fn replay_at_position(&mut self, position: RelativeTime) {
+        let timestamp = position.map_to_absolute_time(&self.frame_range);
         self.time_sender
-            .send(timestamp.inner())
-            .expect("failed to send replay time");
+            .send_modify(|state| state.time = timestamp.inner())
     }
 }
 
 impl App for Window {
     fn update(&mut self, context: &Context, _frame: &mut Frame) {
+        context.input_mut(|input| {
+            if input.consume_key(Modifiers::NONE, Key::Space) {
+                self.time_sender
+                    .send_modify(|state| state.playing = !state.playing);
+            }
+            if input
+                .events
+                .iter()
+                // egui is scheiße
+                .any(|e| *e == Event::Text("<".to_string()))
+            {
+                self.time_sender
+                    .send_modify(|state| state.playback_rate -= 0.25);
+            }
+            if input
+                .events
+                .iter()
+                // egui is scheiße
+                .any(|e| *e == Event::Text(">".to_string()))
+            {
+                self.time_sender
+                    .send_modify(|state| state.playback_rate += 0.25);
+            }
+        });
         CentralPanel::default().show(context, |ui| {
+            ui.label(format!("{}", self.time_sender.borrow().playback_rate));
```

Comment:
This label will claim a lot of horizontal space for a information that is probably not so important to show it all the time. Could you integrate it more intelligently in the UI to save space or maybe show it only shortly? Or in the time? Or with some other visual elements graphically?

Reply:
- I don't think it's a lot of vertical space
- In this UI, horizontal space is a lot more valuable than vertical
- I'd like to add more elements to that row like a button to toggle playing (as an alternative to hotkey), mode selection, etc, which won't make it any higher either.